src/backend/distributed/executor/executor.c

```diff
@@ -499,7 +503,15 @@ StartDistributedExecution(DistributedExecution *execution)
 
 			if (TaskListRequires2PC(taskList))
 			{
+				/*
```

Comment:
FYI: This logic comes from [here](https://github.com/citusdata/citus/blob/febe41210879fd601da52ac4bf7a2178a789d531/src/backend/distributed/executor/multi_router_executor.c#L755-L766)